### PR TITLE
dependencies: CVE scanner for repository_locations.bzl metadata.

### PR DESCRIPTION
**Pull request by htuch**
This is a custom CVE scanner that consumes the NIST CVE database and
heuristically matches against the CPEs, versions and last update stamps
in repository_locations.bzl.

Future PRs will create a CI job that runs this on a periodic basis
(every few hours) to provide a CVE early warning system.

Example output:

Based on heuristic matching with the NIST CVE database, Envoy may be vulnerable to:

  CVE ID: CVE-2019-19391
  CVSS v3 score: 9.1
  Severity: CRITICAL
  Published date: 2019-11-29
  Last modified date: 2019-12-19
  Dependencies: com_github_luajit_luajit
  Description: ** DISPUTED ** In LuaJIT through 2.0.5, as used in Moonjit before
  2.1.2 and other products, debug.getinfo has a type confusion issue
  that leads to arbitrary memory write or read operations, because
  certain cases involving valid stack levels and > options are
  mishandled. NOTE: The LuaJIT project owner states that the debug
  libary is unsafe by definition and that this is not a vulnerability.
  When LuaJIT was originally developed, the expectation was that the
  entire debug library had no security guarantees and thus it made no
  sense to assign CVEs. However, not all users of later LuaJIT
  derivatives share this perspective.
  Affected CPEs:
  - cpe:2.3:a:moonjit_project:moonjit:*
  - cpe:2.3:a:luajit:luajit:*

Risk level: Low
Testing: cve_scan_test.py unit tests, manual.

Signed-off-by: Harvey Tuch <htuch@google.com>

### CODE REVIEW

File: ci/do_ci.sh [hunk 1]
```diff
@@ -412,6 +412,8 @@ elif [[ "$CI_TARGET" == "docs" ]]; then
   # Validate dependency relationships between core/extensions and external deps.
   tools/dependency/validate_test.py
   tools/dependency/validate.py
+  # Validate the CVE scanner works. TODO(htuch): create a dedicated tools CI target.
+  python3.8 tools/dependency/cve_scan_test.py
```
Comment (moderation) [inline ci/do_ci.sh:416]: Does this need the `python3.8 `?

**Comment by moderation** [comment]
Looks great and will need the `last_updated` to `release_date` when ready.

File: ci/do_ci.sh (hunk 1 above)
Comment (htuch) [reply ci/do_ci.sh:416]: Yeah, the datetime methods we use are more recent.

**Comment by htuch** [comment]
@lizan friendly ping on review.

**Review by lizan — APPROVED** [review]

**Pull request merged**